### ISSUE TITLE
[Nightly] Settings are not synced

### ISSUE THREAD

**Issue by rednag** [brave/brave-browser#13555]
## Steps to Reproduce -> Version 1.20.74 Chromium: 88.0.4324.79 nightly (64-Bit)

   1. Add several clients to your sync chain
   2. Activate full sync
   3. All clients will sync bookmars, passwords,... but not the settings of brave

## Actual result:

No screenshot needed.

## Expected result:
-> Sync everything -> Sync everything as well the settings of my browser!

## Reproduces how often: 

Always - I've experienced it when I've installed several Linux test machines and had always set it up in darkmode, address bar, w/o rewards,... although it was sync with my Windows machine

## Brave version (brave://version info)
```
Brave	1.20.74 Chromium: 88.0.4324.79 (Offizieller Build) nightly (64-Bit)
Überarbeitung	bd1e9353659b2491dac971226a973ca3b5684a14-refs/branch-heads/4324@{#1520}
Betriebssystem	Windows 10 OS Version 2004 (Build 19041.685)
JavaScript	V8 8.8.278
User-Agent	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.79 Safari/537.36
Befehlszeile	"C:\Program Files\BraveSoftware\Brave-Browser-Nightly\Application\brave.exe" --enable-dom-distiller --disable-domain-reliability --no-pings --extension-content-verification=enforce_strict --extensions-install-verification=enforce --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --enable-features=WebUIDarkMode,AutoupgradeMixedContent,WinrtGeolocationImplementation,LegacyTLSEnforced,PrefetchPrivacyChanges,PasswordImport,ReducedReferrerGranularity --disable-features=AutofillEnableAccountWalletStorage,TextFragmentAnchor,PrivacySettingsRedesign,NotificationTriggers,WebOTP,PasswordCheck,NetworkTimeServiceQuerying,SafeBrowsingEnhancedProtection,SignedExchangeSubresourcePrefetch,TabHoverCards,AutofillServerCommunication,IdleDetection --flag-switches-begin --flag-switches-end
Pfad zur ausführbaren Datei	C:\Program Files\BraveSoftware\Brave-Browser-Nightly\Application\brave.exe
Profilpfad	C:\Users\rednag\AppData\Local\BraveSoftware\Brave-Browser-Nightly\User Data\Default
Varianten	5e3a236d-59e286d0
```

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Yes
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 
- Yes
## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- No
- Does the issue resolve itself when disabling Brave Rewards? 
- No
- Is the issue reproducible on the latest version of Chrome? 
- No, the settings are brave specific except dark mode.

**Comment by rebron** [comment]
@rednag Sounds like you're asking for Rewards sync.
@jsecretan Is this a dupe? Wasn't sure we actually had a Rewards sync issue filed.

**Comment by rednag** [comment]
Nothing to do with rewards except rewards has to do something with sync?

I'm not able to sync any setting from one device to an other device.

**Comment by jsecretan** [comment]
I think that by default the Chromium sync subsystem only syncs certain settings.  @AlexeyBarabash can you confirm this is true and can you summarize which ones?

**Comment by AlexeyBarabash** [comment]
i can confirm this is true

**Comment by fabiodcorreia** [comment]
I have the same behavior on the stable release. So if I understand this is a limitation on Chromium?

**Comment by jsecretan** [comment]
I believe so @fabiodcorreia.  Now, we could possibly depart with what Chromium does here.  What settings would be most important to sync over in your opinion?

**Comment by fabiodcorreia** [comment]
> I believe so @fabiodcorreia. Now, we could possibly depart with what Chromium does here. What settings would be most important to sync over in your opinion?

Like it was mentioned looks like some settings are already sync, but looking for the settings blocks I think the most important that would be nice setup just one time would be:
- Appearance, specially the parts about the home button, full search bar and bookmarks bar
- Shields
- Social Media Blocking
- Privacy and Security
- Dashboard settings? Not 100% sure about this one, maybe if you use the Brave Today to avoid setting the sources again.

The rewards will also be nice, but I understand that part is probably more complex than just simple settings, I also saw some threads around that subject already.

**Comment by snowbound** [comment]
Seeing similar behavoir on beta builds here

Just installed Brave Beta on a 3rd computer here. Like the other two I have sync set for
Bookmarks, Extensions, History, Settings,Open Tabs.

Once the sync completed on the third Wind10 desktop I noticed the settings were different on the new computer. Hide Brave Rewards button was not enabled nor was Use wide address bar Unlike the existing two computers Allow Twitter embedded tweets, Allow Facebook logins and embedded posts, Allow Google login buttons on third party sites were all enabled on the new computer rather than being disabled as is the case on the two existing computers. Always show full URLs is also not sync'd.

Personally, if this is working as designed I would prefer a very clear indication in the sync configuration screen what exactly in SETTINGS will be sync'd either that or just remove settings altogether from Sync At least that way people are not unnder the disillusion that the settings are consistent across all Brave browsers in the Sync Chain.
.

**Comment by jclark-dot-org** [comment]
Some data points:  I have enabled sync of (Bookmarks, Extensions, Settings, Themes) and it mostly has done exactly what I want, however, several settings under "Appearance" (`brave://settings/appearance`) do not appear to sync; these are the settings I have changed from the defaults which are not syncing:

* Use wide address bar
* Hide Brave Rewards button
* Always Show Full URLs
* Show warning before quitting with ⌘Q
* New Tab Page: Blank

I would expect sync of Settings (or possibly theme?  I don't really use themes) to sync these settings.  By way of comparison, Shields settings (including per-site policy overrides) sync perfectly.

**Comment by c5c86a** [comment]
If this issue is not just for rewards, then the following might help.

**Description of the issue:**

Some settings are synced from desktop to android but not all.

When someone sees the following options in desktop to "Sync data", it is misleading to give the option "Settings", as not all settings get synced.

* Apps
* Bookmarks
* Extensions
* History
* Settings
* Theme
* Open tabs
* Passwords
* Addresses and more

Also the navigation paths of settings in desktop are different from android which is a bit confusing at least when reporting issues :)

**Steps to Reproduce:**

1. On desktop, edit settings in:
    1. Passwords
    2. Payment methods
    3. Addresses
2. On desktop, edit settings in
    1. Search Engines: yes, I love Brave Search!
    2. Social media blocking
    3. Shields: fingerprinting blocking, "trackers and ads blocking"
	4. Additional settings, Privacy and security, "Send a Do Not Track request with your browsing traffic"
3. On desktop go to settings at section Sync, add new device
4. On android go to Settings at Section sync, scan the QR code shown on the desktop screen, go to Data preferences, enable Settings and leave the Settings screen of android (e.g. browse a web page).
5. On desktop go to settings at section Sync, in the Sync Settings below the list of devices synced, enable Settings, leave the Settings page (e.g. browse a web page) and while waiting for a few minutes for syncing to happen, you can check the Traffic Log tab at `brave://sync-internals` or press the "Trigger GetUpdates" in the About tab
6. On android in the settings screen you will see that the sections Passwords, "Payment methods" and "Addresses and more" are synced (step 1) but the sections "Search Engines" and "Brave shields and privacy" of android (all sections of Desktop mentioned in step 2) are not synced.

**Reproduces how often:**

Reproduced this in an Android 10 phone and in another Android 11 phone

**Brave Version:**

V1.27.108 (July 23, 2021)

**Type of devices currently running on the Sync chain in question:**

Windows 10 and android 10 and 11

**Suggestions**:

I assume that developing the capability to sync ALL settings would be hard/impossible so one suggestion is to move all settings that a user can sync now under the same navigation path (and in all device types), for example with root section X, and put X in the options to "Sync data".

If there is any way I can contribute on that, let me know, my background is on QA Automation with API calls and Selenium in python.

**Additional Information:**

I asked also the community [here](https://community.brave.com/t/not-all-settings-are-synced/268748) to see if there can be a more objective set of settings that is worth syncing. There I focus on privacy but I think that the issue is first of all a matter of User Experience (information architecture). 

Once this is resolved then I would suggest treating feature requests of adding support to sync a specific setting on separate tickets.

**Comment by bsclifton** [comment]
+1 from @Germs2004 via https://github.com/brave/brave-browser/issues/18191

With `Settings` enabled in Sync (on both machines; Desktop), it's not syncing specific settings (see issue for details)

**Comment by Germs2004** [comment]
> **Suggestions**:
> 
> I assume that developing the capability to sync ALL settings would be hard/impossible so one suggestion is to move all settings that a user can sync now under the same navigation path (and in all device types), for example with root section X, and put X in the options to "Sync data".

I think this is a great idea.  Put all the syncing settings in one area of the Settings page, and all the non-syncing (per-device) settings in a separate area.  Alternatively, keep them mixed together but put a special icon next to the syncing settings so we know that changing it will affect more than 1 device.

**Comment by thesuperzapper** [comment]
I always assumed they didn't sync hiding the new-tab-page ads and crypto stuff because Brave makes money from those.

That also may be why no one has made an effort to fix this, lol!

**Comment by vdias** [comment]
Any news on this... really annoying to manual change the same settings on all machines... 

Is there any "manual" solution to full export one profile to a different machine?

**Comment by fargozhu** [comment]
Issues opened for years? I don't think I can take Brave as a solid browser.

**Comment by robbycbennett** [comment]
Here's a workaround that worked for me to copy from Windows to Linux. It requires SSH on both and I ran it from Windows CMD. The only setting that I noticed that didn't change was Widevine, which is in "%LocalAppData%\BraveSoftware\Brave-Browser\User Data\Default\Local State" as the key "widevine_opted_in". This disables Sync.

```
scp "%LocalAppData%\BraveSoftware\Brave-Browser\User Data\Default\Preferences" "linux:.config/BraveSoftware/Brave-Browser/Default/"
scp "%LocalAppData%\BraveSoftware\Brave-Browser\User Data\Default\Secure Preferences" "linux:.config/BraveSoftware/Brave-Browser/Default/"
scp "%LocalAppData%\BraveSoftware\Brave-Browser\User Data\Default\Sync Data\LevelDB\053594.ldb" "linux:.config/BraveSoftware/Brave-Browser/Default/Sync Data/LevelDB"
```

Most of the time-consuming settings, like site settings, are copied only with the Preferences file.

I'm not sure about that 053594.ldb file. It may be different for other people.

I would close Brave, then copy this, then open Brave, and finally enable Sync.
